File: 128-longest-consecutive-sequence/longest-consecutive-sequence.py
"""
[0,1,1,2]

"""
class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:
        nums = list(set(nums))
        nums.sort()
        count = 1
        res = 0

        left = 0
        right = 0
        while right < len(nums):
            if left == right:
                count = 1
                right+=1
            while right < len(nums) and nums[right] == nums[right-1] + 1:
                right+=1
            res = max(res, right - left)
            left = right
        return res

            

File: 128-longest-consecutive-sequence/longest-consecutive-sequence.java
class Solution {
    public int longestConsecutive(int[] nums) {
        HashSet<Integer> set = new HashSet<>();
        for(int i = 0;i<nums.length;i++){
            set.add(nums[i]);
        }
        int res = 0;
        for(int num : set){
            if(set.contains(num-1)){
                continue;
            }
            else{
                int curr = num + 1;
                int count = 1;
                while(set.contains(curr)){
                    count+=1;
                    curr+=1;
                }
                res = Math.max(res,count);
            }
        }
        return res;
    }
}